### PR TITLE
NEXT-11875: Fix sorting of order line items

### PR DESCRIPTION
**Pull request by claudiobianco**
### 1. Why is this change necessary?
The order line items are in a different order than the cart line items.

### 2. What does this change do, exactly?
Adds sorting (position) of lineItems association.

### 3. Describe each step to reproduce the issue or behaviour.
Buy some stuff and compare the order of the line items from the cart to the order of the line items of the order.

### 4. Please link to the relevant issues (if any).
https://issues.shopware.com/issues/NEXT-11875
https://issues.shopware.com/issues/NEXT-9869

New PR for #1512 because the source branch got deleted.

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/master/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: changelog/_unreleased/2020-11-16-fix-sorting-of-order-line-items-to-have-the-same-as-in-the-cart.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: Fix sorting of order line items to have the same as in the cart
+issue: NEXT-11875
+author: Claudio Bianco
+author_email: info@claudio-bianco.de 
+author_github: @claudiobianco
+---
+# Core
+*  Add order line items sorting (position) in `CartOrderRoute`, `DocumentService` and `OrderStateChangeEventListener`.
```
Comment (J-Rahe) [inline changelog/_unreleased/2020-11-16-fix-sorting-of-order-line-items-to-have-the-same-as-in-the-cart.md:9]: please change to
* Added order...

**Comment by J-Rahe** [comment]
Hey @claudiobianco, please add some tests.

**Comment by claudiobianco** [comment]
Hey @J-Rahe can you give me a hint what and how to test it?

**Comment by J-Rahe** [comment]
basically add some test that fail if the sorting of the orders is changed. If you look at e.g. the CartOrderRoute.php there is already an CartOrderRouteTest.php for it with an testOrderOneProduct which might help you. To test the sorting you would order multiple items in the same order and check that that order is correct.

**Comment by claudiobianco** [comment]
But the PR does only add the sorting for specific views/use cases. It does not change the sorting in the order per se. So I would have to add the sorting for the test and then check if it is correctly sorted? That would just test if the DAL sorting works correctly? Or am I missing something? ;)

**Comment by claudiobianco** [comment]
Well it would test if the position field is correctly saved from cart line item to the order line item. Would that be sufficant @J-Rahe?

**Comment by J-Rahe** [comment]
Hey @claudiobianco, sorry that i didn't respond before the weekend, what the test should do is to ensure that we will notice when the sorting is changed again. e.g. the test checks that if your order is sorted A,C,B,F with the change, the test would compare the sorting against that.

**Comment by J-Rahe** [comment]
This PR was closed due to inactivity. If this change is still important to you, feel free to create a new pull request.
For more information about our contribution guidelines, see https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline

**Pull request closed without merging**